python/cudf/cudf/testing/_utils.py

```diff
@@ -346,6 +346,11 @@ def get_ptr(x) -> int:
     assert len(lhs.base_children) == len(rhs.base_children)
     for lhs_child, rhs_child in zip(lhs.base_children, rhs.base_children):
         assert_column_memory_eq(lhs_child, rhs_child)
+    if isinstance(lhs, cudf.core.column.CategoricalColumn) and isinstance(
+        rhs, cudf.core.column.CategoricalColumn
+    ):
+        assert_column_memory_eq(lhs.categories, rhs.categories)
+        assert_column_memory_eq(lhs.codes, rhs.codes)
```

Comment:
Why was this change necessary when it wasn't before? What difference does COW make here?

Reply:
Ah got it that makes sense.